rpcq/_server.py

```diff
@@ -74,17 +76,38 @@ def exit_handler(self, f: Callable):
         """
         self._connect(endpoint)
 
-        while True:
-            try:
-                # empty_frame may either be:
-                # 1. a single null frame if the client is a REQ socket
-                # 2. an empty list (ie. no frames) if the client is a DEALER socket
-                identity, *empty_frame, msg = await self._socket.recv_multipart()
-                request = from_msgpack(msg)
+        # spawn an initial listen task
+        listen_task = asyncio.ensure_future(self._socket.recv_multipart())
+        task_list = [listen_task]
 
-                asyncio.ensure_future(self._process_request(identity, empty_frame, request))
-            except Exception:
-                _log.exception('Exception thrown in Server run loop')
+        while True:
+            dones, pendings = await asyncio.wait(task_list, return_when=asyncio.FIRST_COMPLETED)
+            task_list = list(pendings)
+
+            for done in dones:
+                if done == listen_task:
+                    # empty_frame may either be:
+                    # 1. a single null frame if the client is a REQ socket
+                    # 2. an empty list (ie. no frames) if the client is a DEALER socket
+                    identity, *empty_frame, msg = done.result()
+                    request = from_msgpack(msg)
+
+                    # spawn a processing task
+                    task_list.append(asyncio.ensure_future(
+                        self._process_request(identity, empty_frame, request)))
+
+                    # spawn a new listen task
+                    listen_task = asyncio.ensure_future(self._socket.recv_multipart())
```

Comment:
again _I think_ (but am not sure) that the recommended synonym is `asyncio.create_task`